### PR TITLE
Adds IQ chain to dev

### PR DESCRIPTION
**Pull request by Adebesin-Cell**
# Add IQ Chain Integration to Development Environment

This pull request introduces the integration of the IQ chain into the development environment. The changes include the addition of a dynamic configuration to the Wagmi wrapper, along with updates to switch from Polygon Mumbai to IQ testnet. Conditional statements are utilized to determine the appropriate chain to use based on specified conditions. Additionally, modifications have been made to the retrieval of user balances, implementing either the Wagmi hook or a custom hook for Polygon. These adjustments aim to enhance the functionality and flexibility of the development environment by incorporating support for the IQ chain.

Fixes https://github.com/EveripediaNetwork/issues/issues/2583
Fixes https://github.com/EveripediaNetwork/issues/issues/2584
Fixes https://github.com/EveripediaNetwork/issues/issues/2630
Fixes https://github.com/EveripediaNetwork/issues/issues/2650

### CODE REVIEW

**Comment by Adebesin-Cell** [comment]
> Looks good so far.

Good to merge?

**Comment by Adebesin-Cell** [comment]
Observations:

Adding ``ssr: true`` to wagmi config fixes homepage from breaking, but adding that causes edit bug ``connection.connector.getProvider``.Most solutions suggest to turn ssr off tho :))

cc: @Royal-lobster @Tolu-Mals @AshakaE

**Comment by Royal-lobster** [comment]
Getting error at Sent to Relayer

CleanShot 2024-05-06 at 09 59 39@2x

**Comment by Adebesin-Cell** [comment]
> Getting error at Sent to Relayer
> 
> CleanShot 2024-05-06 at 09 59 39@2x

Hm, weird. Prolly gas issues? Though I have a bug with setting SSR to true in config, it makes homepage work which was breaking before but it doesn’t make editing work properly. I was experimenting with dynamic wrapper but create wiki button seemed to stop working.

**Comment by Tolu-Mals** [comment]
Nothing happens when I click publish, also when i refresh the page I get a client side error

**Comment by Tolu-Mals** [comment]
When I click the publish button, nothing happens

https://github.com/EveripediaNetwork/ep-ui/assets/30352484/82f6f0f9-7770-4db1-877d-c5083bc3eaf7

**Comment by Tolu-Mals** [comment]
A modal opened when I refreshed and tried again, but it didn't publish still

https://github.com/EveripediaNetwork/ep-ui/assets/30352484/be6ecca5-741d-4946-8caf-7419a153f880

**Comment by Tolu-Mals** [comment]
We need to change this message depending on the network we're switching to

Screenshot 2024-05-07 at 5 10 27 PM

**Comment by Adebesin-Cell** [comment]
Noted, thx! I'll check them. :good catch 👏

**Comment by Royal-lobster** [comment]
For me getting this error

CleanShot 2024-05-08 at 09 41 39

**Comment by Royal-lobster** [comment]
Also i think switch wallet condition is broken ? cuz its showing to switch network even tho i am on IQ Testnet

CleanShot 2024-05-08 at 09 44 23

**Comment by Royal-lobster** [comment]
Also found this wierd bug, When i go to `/create-wiki`, Switch Chain via modal, Refresh -> it crashes the page

https://github.com/EveripediaNetwork/ep-ui/assets/52039218/97380a24-ff83-4c68-af39-5969a206f2ae

**Comment by Adebesin-Cell** [comment]
> For me getting this error
> 
> CleanShot 2024-05-08 at 09 41 39

logout and login again :))

**Comment by Royal-lobster** [comment]
Ahh ok, logging out and logging in getting this 

https://github.com/EveripediaNetwork/ep-ui/assets/52039218/16647bc9-8b33-47cc-bdd4-7bd38033a768

**Comment by Adebesin-Cell** [comment]
> Visit Preview

seems to be from wallet connect issue again we had for ssr:false or true, but weird, I removed wallet connect now config still throwing same err

**Comment by Royal-lobster** [comment]
Btw, i am able to replicate https://github.com/EveripediaNetwork/issues/issues/2605 issue on this pr

https://github.com/EveripediaNetwork/ep-ui/assets/52039218/eb3ec936-f90c-4bed-a3a3-2a5a4106d7c1

**Comment by Adebesin-Cell** [comment]
> Btw, i am able to replicate [EveripediaNetwork/issues#2605](https://github.com/EveripediaNetwork/issues/issues/2605) issue on this pr
> 
>  CleanShot.2024-05-08.at.10.01.33.mp4

Pls help check if there are errors in console or any request failing in network

**Comment by Royal-lobster** [comment]
> > Visit Preview
> 
> seems to be from wallet connect issue again we had for ssr:false or true, but weird, I removed wallet connect now config still throwing same err

How about we only do ssr:true fixed ? we can test out how it works on preview url

**Comment by Adebesin-Cell** [comment]
> > > Visit Preview
> > 
> > 
> > seems to be from wallet connect issue again we had for ssr:false or true, but weird, I removed wallet connect now config still throwing same err
> 
> How about we only do ssr:true fixed ? we can test out how it works on preview url, i believe the dynamic hack is breaking some - provider swap could cause some issues with wagmi

It stops editing from working, we get back to connection.connector.getProvider is not a function :))

**Comment by Royal-lobster** [comment]
> Pls help check if there are errors in console or any request failing in network

This is console

https://github.com/EveripediaNetwork/ep-ui/assets/52039218/174f3d22-4c3a-4bbf-ad22-135d4fcc9d6a

**Comment by Royal-lobster** [comment]
> network tab?

Seems nothing out of ordinary

https://github.com/EveripediaNetwork/ep-ui/assets/52039218/ec04f044-6f6e-4948-b947-68644da406f3

**Comment by Adebesin-Cell** [comment]
@Royal-lobster I deprecated wallet connect and switched to proper config, kindly test to see if everything works, the next update didn't fix it.

File: src/pages/search/[query].tsx [hunk 1]
```diff
@@ -137,7 +138,17 @@ export const getServerSideProps: GetServerSideProps = async (context) => {
   const query = queryParam as string
 
   return {
-    props: { query },
+    props: {
```
Comment (kesar) [inline src/pages/search/[query].tsx:141]: why there is a fix about search in this PR ? lets open new one w this

File: src/pages/_app.tsx [hunk 2]
```diff
@@ -49,9 +50,11 @@ const App = ({ Component, pageProps, router }: EpAppProps) => {
       <SEOHeader router={router} />
       <ReduxProvider store={store}>
         <ChakraProvider resetCSS theme={chakraTheme}>
-          <Layout noFooter={Component.noFooter}>
-            <Component {...pageProps} />
-          </Layout>
+          <WagmiWrapper>
```
Comment (kesar) [inline src/pages/_app.tsx:53]: can we keep the wrapper as it was for this PR and then do a different one changing this? or this is something related to using a new testnet?

File: src/data/NetworkData.ts [hunk 3]
```diff
@@ -9,10 +9,10 @@ export type Network = {
 export const Networks: Network[] = [
   {
     id: 1,
-    image: '/images/logos/polygon.svg',
-    name: 'Polygon',
+    image: '/images/logos/braindao-logo.svg',
+    name: 'IQ',
     isActive: true,
-    chainId: '0x13881',
+    chainId: '0x333133333737',
```
Comment (kesar) [inline src/data/NetworkData.ts:15]: polygon needs to keep working until we migrate to our own mainnet.

we could change id:2 & 3 to IQ

File: src/pages/search/[query].tsx (hunk 1 above)
Comment (Adebesin-Cell) [reply src/pages/search/[query].tsx:141]: Okay ser, I added it cos I noticed translation for search was missing in footer

File: src/components/CreateWiki/CreateWikiTopBar/WikiPublish/WikiPublishButton.tsx [hunk 4]
```diff
@@ -49,18 +49,20 @@ export const WikiPublishButton = () => {
...
   const { refetch } = useGetEventsQuery({
     offset: 0,
     limit: EVENT_TEST_ITEM_PER_PAGE,
   })
 
-  const { chainId } =
-    config.alchemyChain === 'maticmum'
-      ? networkMap.MUMBAI_TESTNET
-      : networkMap.POLYGON_MAINNET
+  const { chainId } = JSON.parse(config.isProduction)
```
Comment (kesar) [inline src/components/CreateWiki/CreateWikiTopBar/WikiPublish/WikiPublishButton.tsx:62]: why config.isProduction is somehow a json? shouldnt be this a boolean?

File: src/data/NetworkData.ts (hunk 3 above)
Comment (Adebesin-Cell) [reply src/data/NetworkData.ts:15]: Ha, good catch 👏

File: src/components/CreateWiki/CreateWikiTopBar/WikiPublish/WikiPublishButton.tsx (hunk 4 above)
Comment (Adebesin-Cell) [reply src/components/CreateWiki/CreateWikiTopBar/WikiPublish/WikiPublishButton.tsx:62]: It appears stringified so I'm using Json.parse to turn it to boolean :))

value is "true" the string

Comment (Adebesin-Cell) [reply src/components/CreateWiki/CreateWikiTopBar/WikiPublish/WikiPublishButton.tsx:62]: Noted ser, I'll fix it to make it more readable. I think updating envs would be better :))
Thanks ser.

**Comment by Royal-lobster** [comment]
@Adebesin-Cell Its not showing my IQ balance correctly

CleanShot 2024-05-10 at 14 21 39@2x

CleanShot 2024-05-10 at 14 21 44@2x

**Comment by Adebesin-Cell** [comment]
> @Adebesin-Cell Its not showing my IQ balance correctly
> 
> CleanShot 2024-05-10 at 14 21 39@2x
> 
> CleanShot 2024-05-10 at 14 21 44@2x

	Can you paste me your address so I can use in testing?

**Comment by Royal-lobster** [comment]
Also fails at relayer

**Comment by Royal-lobster** [comment]
> Can you paste me your address so I can use in testing?

0x1dfc530a9b3955d62d16359110e3cf385d47b1a9

**Comment by Adebesin-Cell** [comment]
@Royal-lobster Image not showing for the relayer fail. @AshakaE can you help check this out?

**Comment by Adebesin-Cell** [comment]
> > Can you paste me your address so I can use in testing?
> 
> 0x1dfc530a9b3955d62d16359110e3cf385d47b1a9

thx

**Comment by Adebesin-Cell** [comment]
https://github.com/EveripediaNetwork/ep-ui/assets/66301634/276b1-b06a-4cd1-94cd-a11e84998c42




@Royal-lobster works for me, same url

**Comment by Royal-lobster** [comment]
For me :/

https://github.com/EveripediaNetwork/ep-ui/assets/52039218/4bad38a9-61fe-46aa-9c4c-175105459f2a

**Comment by Royal-lobster** [comment]
Also i am getting this randomly. won't go away unless i clear my cache. even after clearing it comes back after some time

CleanShot 2024-05-10 at 14 38 06@2x

**Comment by Adebesin-Cell** [comment]
> For me :/
> 
>  CleanShot.2024-05-10.at.14.31.28.mp4

Weird, really weird. Relayer issue should prolly stem from low gas issues or sth, maybe check network tab and send what error the graphQL gives?

**Comment by Adebesin-Cell** [comment]
https://github.com/EveripediaNetwork/ep-ui/assets/66301634/e85be2b5-fe4b-4a62-8675-6040d6f3cec9

chrome too

File: src/pages/search/[query].tsx (hunk 1 above)
Comment (kesar) [reply src/pages/search/[query].tsx:141]: so please create a small PR related to that and remove from here. This PR has already enough things to test to add random stuff 👍🏻

File: src/data/WalletData.ts [hunk 5]
```diff
@@ -1,6 +1,6 @@
 export const walletsLogos: string[] = [
   'metamask.png',
-  'walletconnect-logo.svg',
+  // 'walletconnect-logo.svg',
```
Comment (kesar) [inline src/data/WalletData.ts:3]: remove and remove file

File: src/env.mjs [hunk 6]
```diff
@@ -14,7 +14,7 @@ const server = z.object({
 })
 
 const client = z.object({
-  NEXT_PUBLIC_IS_PRODUCTION: z.string(),
+  NEXT_PUBLIC_IS_PRODUCTION: z.preprocess((v) => v === 'true', z.boolean()),
```
Comment (kesar) [inline src/env.mjs:17]: nice one :)

File: src/hooks/UseFetchWallet.tsx [hunk 7]
```diff
@@ -1,17 +1,16 @@
...
+import config from '@/config'
+import { useAccount, useBalance } from 'wagmi'
 
 export const getUserIQBalance = async (userAddress: string) => {
   try {
     const response = await axios.post<{ result: string }>(
-      `https://eth-${
-        env.NEXT_PUBLIC_IS_PRODUCTION === 'true' ? 'mainnet' : 'goerli'
-      }.alchemyapi.io/v2/${env.NEXT_PUBLIC_ALCHEMY_API_KEY}`,
+      `https://eth-mainnet.alchemyapi.io/v2/${env.NEXT_PUBLIC_ALCHEMY_API_KEY}`,
```
Comment (kesar) [inline src/hooks/UseFetchWallet.tsx:13]: not checking here if its iq testnet will always return stuff from mainnet

File: src/hooks/UseFetchWallet.tsx [hunk 8]
```diff
@@ -43,9 +42,13 @@ export const getUserIQBalance = async (userAddress: string) => {
   }
 }
 
-export const useFetchWalletBalance = (address: string | null) => {
+export const useFetchWalletBalance = () => {
   const [userBalance, setUserBalance] = useState<WalletBalanceType[]>([])
   const dispatch = useDispatch()
+  const { address } = useAccount()
+  const { data, isLoading: isBalanceLoading } = useBalance({
```
Comment (kesar) [inline src/hooks/UseFetchWallet.tsx:49]: i dont fully understand this piece of code. are we getting data balance but using getUserIQBalance() if its production. and getUserIQBalance only gets IQ balance if its production but wouldnt work if its testnet? 

I think getUserIQBalance should get user iq balance and the implementation of where its getting result (testnet or mainnet) should be inside of that.

File: src/hooks/useWeb3Token.ts [hunk 9]
```diff
@@ -31,12 +31,12 @@ export const useWeb3Token = () => {
         (msg) => walletClient.signMessage({ message: msg }),
         {
           statement:
-            'Welcome to IQ.GPT ! Click to sign in and accept the IQ.GPT Terms of Service. This request will not trigger a blockchain transaction or cost any gas fees. Your authentication status will reset after 7 days.',
-          expires_in: '7d',
+            'Welcome to IQ.GPT ! Click to sign in and accept the IQ.GPT Terms of Service. This request will not trigger a blockchain transaction or cost any gas fees. Your authentication status will reset after 365 days.',
```
Comment (kesar) [inline src/hooks/useWeb3Token.ts:34]: Probably copy/paste error. Here should say IQ.wiki

File: src/pages/_app.tsx [hunk 10]
```diff
@@ -35,6 +36,10 @@ const App = ({ Component, pageProps, router }: EpAppProps) => {
     return () => router.events.off('routeChangeComplete', handleRouteChange)
   }, [router.events])
 
+  const isCreateWikiPath = router.pathname === '/create-wiki'
+  const ssr = !isCreateWikiPath
```
Comment (kesar) [inline src/pages/_app.tsx:40]: this doesnt seem to belong to app. most likely pass router.pathname to createWagmiWrapper and do the checks there. 

and test this carefully because this piece of code looks strange to me and i feel you are patching an issue you found on create-wiki that can be in many other pages.

Comment (Adebesin-Cell) [reply src/pages/_app.tsx:40]: Oh dammit, I forgot to remove it. It’s junk

File: src/hooks/UseFetchWallet.tsx (hunk 7 above)
Comment (Adebesin-Cell) [reply src/hooks/UseFetchWallet.tsx:13]: I think we can remove the function and stick to useIQbalance

**Comment by Adebesin-Cell** [comment]
@Royal-lobster 
Screenshot 2024-05-10 at 12 42 59 PM
works!

File: src/hooks/useWeb3Token.ts (hunk 9 above)
Comment (Adebesin-Cell) [reply src/hooks/useWeb3Token.ts]: Fixed!

File: src/hooks/UseFetchWallet.tsx (hunk 8 above)
Comment (Adebesin-Cell) [reply src/hooks/UseFetchWallet.tsx:11]: Refactored to use only a hook

**Comment by Royal-lobster** [comment]
> @Royal-lobster Screenshot 2024-05-10 at 12 42 59 PM works!


Hmm Still same :/ not working for me

CleanShot 2024-05-14 at 09 13 43


CleanShot 2024-05-14 at 09 13 49

**Comment by SidharthK2** [comment]
> > @Royal-lobster Screenshot 2024-05-10 at 12 42 59 PM works!
> 
> Hmm Still same :/ not working for me
> 
> CleanShot 2024-05-14 at 09 13 43
> 
> CleanShot 2024-05-14 at 09 13 49

Should be fixed now

**Comment by Royal-lobster** [comment]
Getting this error 

CleanShot 2024-05-21 at 11 08 01@2x

**Comment by Royal-lobster** [comment]
Hmm it went away after refresh... ill see if i can reproduce it again, also this should be IQ Testnet i think

CleanShot 2024-05-21 at 11 08 58@2x

**Comment by SidharthK2** [comment]
> Getting this error
> 
> CleanShot 2024-05-21 at 11 08 01@2x

image
I'm also getting this error sometimes, only on dev.

**Comment by Royal-lobster** [comment]
Ahh thats wierd, i think its coming from wagmi cuz now we are wrapping the entire app as per docs as well, let me see

**Comment by Royal-lobster** [comment]
Also, this message should only show on last step, may be something broke there cuz its showing for all 3 steps

CleanShot 2024-05-21 at 11 16 27@2x

**Comment by Royal-lobster** [comment]
And for me it keeps on loading in last step, but when i refresh, i can see the wiki is indeed created, may be the code for where we are checking the transaction status broke

CleanShot 2024-05-21 at 11 18 59@2x

**Comment by Royal-lobster** [comment]
@SidharthK2 once try upgrading the wagmi package, probably it might fix the issue

CleanShot 2024-05-21 at 11 21 57@2x

CleanShot 2024-05-21 at 11 47 36@2x

**Comment by SidharthK2** [comment]
> @SidharthK2 once try upgrading the wagmi package, probably it might fix the issue
> 
> CleanShot 2024-05-21 at 11 21 57@2x
> 
> CleanShot 2024-05-21 at 11 47 36@2x

Error seems to be gone after removing early return from useWhitelistValidator hook

**Comment by SidharthK2** [comment]
> And for me it keeps on loading in last step, but when i refresh, i can see the wiki is indeed created, may be the code for where we are checking the transaction status broke
> 
> CleanShot 2024-05-21 at 11 18 59@2x

I think the delay is due to long block time on testnet, I reduced the block confirmations from 2 -> 1 and it's working fine.
image

**Review by Softdev1 — APPROVED** [review]
I tested this issue with the following test cases and it passed 
- [ ]  User should be able to see the wallet balances on the dev. env. 
- [ ]  User should be able to Publish both edited and newly created wikis on the dev. env. 
- [ ]  Only whitelisted addresses should publish wikis. 
- [ ] User should be able to switch network from mainet to IQ testnet without any issues. 
- [ ] User should be able to sign wiki without any error on the dev. evn. 
- [ ] User should see the right network logo should be picked on the drawer


image
image

**Review by Tolu-Mals — APPROVED** [review]

File: src/components/Blog/BlogPost.tsx [hunk 11]
```diff
@@ -13,7 +13,7 @@ export const BlogPost = (props: BlogPostType) => {
   const { post, ...rest } = props
 
   return (
-    <LinkWrapper href={`/blog/${post.digest}`}>
+    <LinkWrapper href={`/blog/${post?.digest}`}>
```
Comment (kesar) [inline src/components/Blog/BlogPost.tsx:16]: a PR of 600 LoC changes shouldnt include unrelated stuff like blog post changes. Not needed to rollback, but I already mentioned several times. If something is blocking an issue, you can create a small PR to fix that issue ( blog issue on dev environment ) and then work in the real issue ( IQ dev chain for example )

File: src/components/CreateWiki/CreateWikiTopBar/WikiPublish/WikiPublishButton.tsx [hunk 12]
```diff
@@ -148,7 +149,7 @@ export const WikiPublishButton = () => {
         )
       }
     }
-  }, [detectedProvider, isUserConnected])
+  }, [detectedProvider, userAddress])
```
Comment (kesar) [inline src/components/CreateWiki/CreateWikiTopBar/WikiPublish/WikiPublishButton.tsx:152]: if isConnected is removed (not sure why since that shouldnt affect switching mumbai to IQ testnet), then we should not initialize it on line 50, I think we are not using it anywhere else

File: src/components/Layout/Network/NetworkErrorNotification.tsx [hunk 13]
```diff
@@ -115,8 +117,9 @@ const NetworkErrorNotification = ({
           </Flex>
           <Text mt="6" w="90%" lineHeight="2">
             Your wallet is currently connected to an unsupported network. To
-            continue with Polygon network, Switch the network in your wallet to
-            Polygon.
+            continue with {isProduction ? 'Polygon' : 'IQ chain'} network,
+            Switch the network in your wallet to{' '}
+            {isProduction ? 'Polygon' : 'IQ chain'}.
```
Comment (kesar) [inline src/components/Layout/Network/NetworkErrorNotification.tsx:122]: 2 ternaries and a const definition looks a bit unecesary for a sentence. Not a big deal... but this could be a bit more simple even if we just copy few extra words twice. Or even define it as chainName and just use that outside of the jsx

File: src/config/wagmi.ts [hunk 14]
```diff
@@ -1,35 +1,47 @@
...
   transports: {
     [polygon.id]: fallback([
       http(`https://polygon-mainnet.g.alchemy.com/v2/${config.alchemyApiKey}`),
       http(`https://polygon-mainnet.infura.io/v3/${config.infuraId}`),
     ]),
-    [polygonMumbai.id]: fallback([
-      http(`https://polygon-mumbai.g.alchemy.com/v2/${config.alchemyApiKey}`),
-      http(`https://polygon-mumbai.infura.io/v3/${config.infuraId}`),
-    ]),
+    [iqTestnet.id]: http(),
```
Comment (kesar) [inline src/config/wagmi.ts:41]: is this correct? shouldnt we put there rpc of iqtestnet?

File: src/components/Blog/BlogPost.tsx (hunk 11 above)
Comment (SidharthK2) [reply src/components/Blog/BlogPost.tsx:16]: Noted ser

File: src/components/CreateWiki/CreateWikiTopBar/WikiPublish/WikiPublishButton.tsx (hunk 12 above)
Comment (SidharthK2) [reply src/components/CreateWiki/CreateWikiTopBar/WikiPublish/WikiPublishButton.tsx:152]: We're using it here to check if MM is locked
image

File: src/config/wagmi.ts (hunk 14 above)
Comment (SidharthK2) [reply src/config/wagmi.ts:41]: It defaults to RPC provided in chain definition

**Comment by Adebesin-Cell** [comment]
It does passes the 3 steps, says wiki published and a confetti, but no sign of it, I did notice an issue with internal activity channel getting a lot of logs prolly some days after edit/create and it seems to be related with the bug atm.

**Comment by AshakaE** [comment]
> @kesar The pr was tested for edit and creating it works fine but it seems there's an issue with revalidation, activity page won't update and history also for the edit or created wiki.
> 
> cc: @AshakaE

Dev indexer was paused 😅, all edits should come through now

**Comment by Adebesin-Cell** [comment]
> > @kesar The pr was tested for edit and creating it works fine but it seems there's an issue with revalidation, activity page won't update and history also for the edit or created wiki.
> > cc: @AshakaE
> 
> Dev indexer was paused 😅, all edits should come through now

Sweet! Let's merge tmr then and watch out for any issues
